src/dag/dag.cpp

```diff
@@ -443,9 +442,9 @@ void DagManager::addToDag(blk_hash_t const &hash, blk_hash_t const &pivot, std::
 
 bool DagManager::getLatestPivotAndTips(blk_hash_t &pivot, std::vector<blk_hash_t> &tips) const {
   // make sure the state of dag is the same when collection pivot and tips
-  sharedLock lock(mutex_);
   pivot.clear();
   tips.clear();
+  sharedLock lock(mutex_);
```

Comment:
- It seems like this lock is protecting data that are read inside getFrontier() function -> shouldn't be the lock there ? Or is this some sort of optimization that we dont want to use locks for every call to getFrontier ? 
- as sharedLock is alias to type, it should be with capital first letter

Reply:
getFrontier is used also in different function, that is under uniqe lock, so I did not changed how we are locking data. 
and I will rename it for dag.cpp